src/check/resolve_imports/IR.zig

```diff
@@ -85,19 +85,19 @@ pub const DeclarationTag = union(enum) {
 };
 
 /// Marks whether a recursive let-cycle was determined to be illegal during solving.
-pub const IllegalCycleMark = ?Type.Idx;
+pub const IllegalCycleMark = ?TypeVar;
```

Comment:
Now that defs are only allowed to be recursive if they're lambdas, I think mutually recursive functions that call each other back and forth infinitely are the only remaining way this can occur. Something like:

```elixir
foo = |x| bar(x)
bar = |x| foo(x)
```

Since this is an error type now, how about if we rename it from `IllegalCycleMark` to the error that will be reported to the programmer in this situation, e.g. `InfiniteMutualRecursion`?